### PR TITLE
Add pendingExpirationTimestamp information

### PR DESCRIPTION
**Pull request by ubbabeck**
# About pr

Implements  and depends on https://github.com/breez/breez-sdk/pull/674

- Adds `pendingExpirationBlock` to the PaymentMinutiae.

- Creates function to convert block difference to time.

I decided to have to pendingExpirationBlock in the payment field and not the calculated approximate timestamp as we might want to use the `pendingExpirationBlock` for something else at a later stage. Let me know if we'd rather want to add `pendingExpirationTime` to PaymentMinutiae instead.

### CODE REVIEW

File: lib/routes/home/widgets/payments_list/dialog/payment_details_dialog_expiration.dart [hunk 1]
```diff
@@ -39,7 +44,21 @@ class PaymentDetailsDialogExpiration extends StatelessWidget {
...
+                pendingExpirationTime != null
+                    ? BreezDateUtils.formatYearMonthDayHourMinute(pendingExpirationTime)
+                    : "",
+                style: themeData.primaryTextTheme.displaySmall,
+                textAlign: TextAlign.right,
+                maxLines: 1,
+              ),
+            ),
+          )
         ],
```
Comment (erdemyerebasmaz) [inline lib/routes/home/widgets/payments_list/dialog/payment_details_dialog_expiration.dart:62]: Maybe it's better to use a helper method for pending expiration time so the logic is more compact.

```
AutoSizeText(
    _getExpirationTime,
    ...
...
_getExpirationTime() {
    final blockHeight = context.read<AccountBloc>().state.blockheight;
    final pendingExpirationBlock = paymentMinutiae.pendingExpirationBlock;
    if(pendingExpirationBlock != null) {
        final pendingExpirationTime = BreezDateUtils.blockDiffToDate(
            blockHeight: blockHeight,
            expiryBlock: pendingExpirationBlock,
        );
        return BreezDateUtils.formatYearMonthDayHourMinute(pendingExpirationTime);
    }
   return "";
}
```

File: lib/utils/date.dart [hunk 2]
```diff
@@ -39,6 +39,18 @@ class BreezDateUtils {
...
+      int diff = expiryBlock - blockHeight;
+      if (diff > 0) {
+        diff *= 600;
+        final time = DateTime.now();
+        return time.add(Duration(seconds: diff));
+      }
+    }
+    return null;
+  }
+
```
Comment (erdemyerebasmaz) [inline lib/utils/date.dart:53]: Let's make this non-nullable.
```suggestion
  static DateTime blockDiffToDate({required int blockHeight, required int expiryBlock}) {
      int diff = expiryBlock - blockHeight;
      if (diff > 0) {
        diff *= 600;
        final time = DateTime.now();
        return time.add(Duration(seconds: diff));
      }
  }

```

**Review by erdemyerebasmaz — COMMENTED** [review]
Looks good
Have 2 nit comments

File: lib/utils/date.dart (hunk 2 above)
Comment (erdemyerebasmaz) [reply lib/utils/date.dart:53]: I suppose it should return current date if there's no difference.

File: lib/routes/home/widgets/payments_list/dialog/payment_details_dialog_expiration.dart (hunk 1 above)
Comment (ubbabeck) [reply lib/routes/home/widgets/payments_list/dialog/payment_details_dialog_expiration.dart:58]: Removed this logic and added a `pendingExpirationTime` to the PaymentMinutiae also.

File: lib/utils/date.dart [hunk 3]
```diff
@@ -40,15 +40,11 @@ class BreezDateUtils {
...
-        final time = DateTime.now();
-        return time.add(Duration(seconds: diff));
-      }
-    }
-    return null;
+    if (expiryBlock == null || expiryBlock <= blockHeight) return null;
+    int diff = expiryBlock - blockHeight;
+    diff *= 600;
+    final time = DateTime.now();
+    return time.add(Duration(seconds: diff));
```
Comment (erdemyerebasmaz) [inline lib/utils/date.dart:47]: Avoid redundant calculation for `diff` variable.
```suggestion
    int diffInSecs = (expiryBlock - blockHeight) * 600;
    final time = DateTime.now();
    return time.add(Duration(seconds: diffInSecs));
```

File: lib/routes/home/widgets/payments_list/payment_item_amount.dart [hunk 4]
```diff
@@ -97,105 +97,109 @@ void main() async {
...
+                        keysend: false,
+                        bolt11: "",
+                      ),
                     ),
                   ),
-                ),
-                getSystemAppLocalizations(),
-              ),
+                  getSystemAppLocalizations(),
+                  800000),
```
Comment (erdemyerebasmaz) [inline lib/routes/home/widgets/payments_list/payment_item_amount.dart:148]: Can you add a comma after these values and reformat code? It helps with readability & keeping the right indentation.

File: lib/utils/date.dart (hunk 2 above)
Comment (ubbabeck) [reply lib/utils/date.dart]: Done

File: lib/routes/home/widgets/payments_list/payment_item_amount.dart (hunk 4 above)
Comment (ubbabeck) [reply lib/routes/home/widgets/payments_list/payment_item_amount.dart]: Added

File: lib/utils/date.dart (hunk 3 above)
Comment (ubbabeck) [reply lib/utils/date.dart]: Added

**Review by ademar111190 — APPROVED** [review]

**Comment by roeierez** [comment]
@ubbabeck @erdemyerebasmaz let's move this forward, seems like not much (if any )work left.

**Review by erdemyerebasmaz — APPROVED** [review]
LGTM
Please address CI issue and the typo before merging.

**Pull request merged**